### PR TITLE
[MINOR][BUILD] Fix about 15 misc build warnings

### PR DESCRIPTION
**Pull request by srowen**
### What changes were proposed in this pull request?

This addresses about 15 miscellaneous warnings that appear in the current build.

### Why are the changes needed?

No functional changes, it just slightly reduces the amount of extra warning output.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Existing tests, run manually.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/deploy/history/HistoryServerSuite.scala [hunk 1]
```diff
@@ -366,7 +366,7 @@ class HistoryServerSuite extends SparkFunSuite with BeforeAndAfter with Matchers
     server.attachHandler(contextHandler)
 
     implicit val webDriver: WebDriver =
-      new HtmlUnitDriver(BrowserVersion.INTERNET_EXPLORER_11, true)
+      new HtmlUnitDriver(BrowserVersion.INTERNET_EXPLORER, true)
```
Comment (MaxGekk) [inline core/src/test/scala/org/apache/spark/deploy/history/HistoryServerSuite.scala]: Just wondering why Internet Explorer at all. Why no `BrowserVersion.getDefault()`?

Comment (srowen) [reply core/src/test/scala/org/apache/spark/deploy/history/HistoryServerSuite.scala]: https://github.com/apache/spark/pull/24843#discussion_r294049535 is relevant, but, I imagine any modern browser would handle the jquery JS fine too. I can try just removing this specification, as it (now) defaults to Chrome (51). I think before it emulated IE 8.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending Jenkins).

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @srowen and @MaxGekk .

**Comment by dongjoon-hyun** [comment]
Some of them looks relevant to `branch-2.4`, too.
Could you make a backporting PR please, @srowen ?

**Comment by srowen** [comment]
Sure, only a handful affected 2.4: https://github.com/apache/spark/pull/25857

**Pull request closed without merging**